### DISCUSSION TITLE
impl `ValueType` for `DateTime<Tz>`

### DISCUSSION THREAD

**Discussion by MuhannadAlrusayni** [SeaQL/sea-query#134] [Q&A] [answered]
Hi, while trying to use `DateTime<Uts>` as column type in my model, I am blocked since `ValueType` is not implemented for `DateTime<Utc>`.

SeaQuery already impl `From<DateTime<Tz>> for Value`, so I think we should have `impl ValueType for DateTime<Tz>` too.

**Comment by tyt2y3** [comment, marked as answer]
`DateTime<FixedOffset>` is supported for now.
May be we can support `DateTime<Utc>` in addition to that, as a separate type, but it will be quite difficult to support other `DateTime<Tz>`.

**Comment by MuhannadAlrusayni** [comment]
Oh, didn't know this is the case, to me I have switched to `DateTime<FixedOffset>` not big deal.

From an API consistency point of view, I'm not sure if it's right to support `DateTime<Utc>` only or just left it like this or maybe remove the blanket `impl From<DateTime<Tz>> for Value`.

It doesn't feels right to be able to convert `DateTime<Tz>` to `Value` but not the other way.

**Comment by tyt2y3** [comment]
> It doesn't feels right to be able to convert DateTime<Tz> to Value but not the other way.

I can explain. Because given any `Tz` of `DateTime<Tz>` we can actually convert it to a `FixedOffset` but not the other way around.
For example, the Local time could be UTC, and it's also a fixed offset of 0.
